### DISCUSSION TITLE
build error when using incorrect version of ESP-IDF

### DISCUSSION THREAD

**Discussion by FreemanGBG** [Moddable-OpenSource/moddable#649] [General]
I was just getting started on Moddable Two and doing some basic examples. Here is what I currently face:

C:\Projects\moddable\examples\helloworld>mcconfig -d -m -p esp32/moddable_two
C:\Projects\moddable\build\tmp\esp32\moddable_two\debug\helloworld\makefile(216) : fatal error U1050: Detected ESP-IDF version v4.2.1. Expected ESP-IDF version v4.2.
Stop.

I checked espressif/esp-idf (https://github.com/espressif/esp-idf/tree/release/v4.2) that they actually update v4.2 to v4.2.1 a month ago, and there is no available v4.2 so far. I know there might be some version check in makefiles.

Any solution?

**Comment by phoddie** [comment]
That's annoying. Assuming there are no incompatible changes, you should be able to change the version check. In `make.esp32.mk` on macOS/Linux or `nmake.esp32.mk ` on Windows, change the `EXPECTED_IDF` from 4.2 to 4.2.1:

```
EXPECTED_ESP_IDF ?= v4.2.1
```

**Reply by FreemanGBG** [reply]
So far it works great. It seems like all changes are compatible.

**Reply by phoddie** [reply]
Cool. Thanks for confirming. Seems to work here too. We'll make that change for our next SDK update.
